lib/util.js

```diff
@@ -732,11 +749,13 @@ module.exports = {
   // and generate the body accordingly
   // right now, even if the content-type was XML, we'll generate
   // a JSON example/schema
-  convertToPmBodyData: function(bodyObj, contentType, bodyTypeOption, indentCharacter = '') {
+  convertToPmBodyData: function(bodyObj, requestType, contentType, bodyTypeOption, indentCharacter = '') {
     var bodyData = '',
-      schemaType = SCHEMA_FORMATS.DEFAULT;
+      schemaType = SCHEMA_FORMATS.DEFAULT,
+      resolveTo = '';
 
-    if (bodyObj.example) {
+    if (bodyObj.example && (this.options.requestParametersResolution === 'example' ||
```

Comment:
Can this be simplified a bit? This is in direct violation of item #5 in the `code` section of our internal guidelines